### DISCUSSION TITLE
How to `add_defines` on a specific file?

### DISCUSSION THREAD

**Discussion by davidchisnall** [xmake-io/xmake#3617] [Q&A] [answered]
The docs for `add_defines` tell me how to do it at a target scope.  If I have a target that builds multiple files and I want to add a define to some of those files but not all of them, how do I do that?

**Comment by waruqi** [comment, marked as answer]
```
add_files("src/*.c", {defines = "xxx"})
```

**Comment by william9523** [comment]
Hey, 

How can I add two macros for a translation unit? 

Tried this one, but does not work. The two macros are combined as one single macro.

add_files("src/*.c", {defines = "VALUE1=1, VALUE2=2"})

**Reply by waruqi** [reply]
use table

```
add_files("src/*.c", {defines = {"VALUE1=1", "VALUE2=2"}})
```
